src/viewer.js

```diff
@@ -320,32 +398,245 @@ function _coordinateFormatScoord3d2Geometry(coordinates, pyramid) {
+    _updateFeatureMeasurements(map, feature);
+  });
+
+  /**
+   * Update feature evaluations
+   */
+  _updateFeatureEvaluations(feature);
+  feature.on(Enums.FeatureEvents.PROPERTY_CHANGE, () =>
+    _updateFeatureEvaluations(feature)
+  );
+}
+
+/**
+ * Update feature evaluations from its properties
+ *
+ * @param {Feature} feature The feature
+ * @returns {void}
+ */
+function _updateFeatureEvaluations(feature) {
+  const label = feature.get(Enums.InternalProperties.Label);
+  const evaluations = feature.get(Enums.InternalProperties.Evaluations) || [];
+
+  if (!label) return;
+
+  const nameCodedConceptValue = "112039";
+  const nameCodedConceptMeaning = "Tracking Identifier";
+  const evaluation = getTextEvaluationContentItem(
+    label,
+    nameCodedConceptValue,
+    nameCodedConceptMeaning
+  );
+
+  const index = evaluations.findIndex((e) =>
+    isContentItemsEqual(e, evaluation)
+  );
+
+  if (index > -1) {
+    evaluations[index] = evaluation;
+  } else {
+    evaluations.push(evaluation);
+  }
+
+  feature.set(Enums.InternalProperties.Evaluations, evaluations);
+  console.debug(`Evaluations of feature (${feature.getId()}):`, evaluations);
+}
+
+/**
+ * Updates feature measurement properties
+ *
+ * @param {object} map The map instance
+ * @param {object} feature The feature instance
+ * @returns {void}
+ */
+const _updateFeatureMeasurementProperties = (map, feature) => {
+  /**
+   * Open Layers side-effect: Geometry will be changed
+   * inside the formCircle call which causes errors
+   * if const variable is used for geometry
+   */
+  let geometry = feature.getGeometry().clone();
+  if (geometry instanceof LineStringGeometry) {
+    feature.set(Enums.FeatureMeasurement.Length, getLength(geometry));
+  } else if (geometry instanceof CircleGeometry) {
+    geometry = fromCircle(geometry);
+    feature.set(Enums.FeatureMeasurement.Area, getArea(geometry));
+  } else if (geometry instanceof PolygonGeometry) {
+    feature.set(Enums.FeatureMeasurement.Area, getArea(geometry));
+  }
+};
+
+/**
+ * Generate feature measurements from its measurement properties
+ *
+ * @param {object} map The map instance
+ * @param {object} feature The feature instance
+ * @returns {void}
  */
-function _getROIFromFeature(feature, pyramid) {
-  if (feature !== undefined && feature !== null) {
-    const geometry = feature.getGeometry();
-    const scoord3d = _geometry2Scoord3d(geometry, pyramid);
-    const properties = feature.getProperties();
-    // Remove geometry from properties mapping
-    const geometryName = feature.getGeometryName();
-    delete properties[geometryName];
-    const uid = feature.getId();
-    return new ROI({ scoord3d, properties, uid });
-  }
-  return
+function _updateFeatureMeasurements(map, feature) {
+  const measurements = feature.get(Enums.InternalProperties.Measurements) || [];
+  const area = feature.get(Enums.FeatureMeasurement.Area);
+  const length = feature.get(Enums.FeatureMeasurement.Length);
```

Comment:
I am wondering whether we should hard-code "Area" and "Length" here.

Could we extend the enum logic?

```js
export const FeatureMeasurement = {
  Area: {
    key: "area",
    code: {CodeValue: "42798000", CodeMeaning: "Area", CodingSchemeDesignator: "SCT"}
  },  
  Length: {
    key: "length",
    code: {CodeValue: "410668003", CodeMeaning: "Length", CodingSchemeDesignator: "SCT"}
  }
};
```

This would allow us to just loop over measurements, extract the measured value from `feature` and construct the SR content item.